lib/pharos/ssh/tempfile.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+require 'securerandom'
+
+module Pharos
+  module SSH
+    # A temporary filename on a remote host
+    # Optionally uploads given content.
+    # When used with a block, removes the temporary file after execution.
+    class Tempfile
+      attr_reader :path
+      # @param client [Pharos::SSH::Client]
+      # @param prefix [String] Filename prefix, default "pharos"
+      # @param content [NilClass,String,IO] Content to upload to remote host
+      # @param file [NilClass,String] Path to a file to upload to remote host
+      # @yield [String] Temporary filename
+      # @example
+      #   Pharos::SSH::Tempfile.new(client) do |path|
+      #     client.exec("uname -a >> #{path}")
+      #   end
+      # @example
+      #   tempfile = Pharos::SSH::Tempfile.new(client)
+      #   client.exec("uname -a >> #{path}")
+      #   tempfile.unlink
+      def initialize(client, prefix: "pharos", content: nil, file: nil, &block)
+        @client = client
+        @path = File.join('/tmp', "#{prefix}.#{SecureRandom.hex(16)}")
+        raise ArgumentError, "Both file and content given" if content && file
+        upload_content(content) if content
+        upload_file(file) if file
```

Comment:
The tempfile could be integrated in the remotefile thing now as they are in one package.

Reply:
Yeah, I guess it can go either way, and it vaugely makes sense that `file.copy(foo)` and `file.link(foo)` both cause a new file to be created at `foo`.

Needs yardoc to clarify.